app/controllers/notes_controller.rb

```diff
@@ -44,6 +44,29 @@ def raw
     render plain: Node.find(params[:id]).latest&.body
   end
 
+  # takes an activerecord query, returns a plain array
+  def get_react_comments(comments_record, get_replies = false)
```

Comment:
We may want to take a static version of the output from here and use it in a test; this could lay groundwork for externalizing the JS code in a library and then the JSON this generates would be the agreed-upon format for exchange between Rails and the JS lib; tested on both ends: here we'd test that Rails correctly generates the right format, while on the JS end, we'd test that the given format is readable and usable by the code. 

We can also annotate the JSON properties required in the README of the JS library, to explain what the parent app is supposed to supply! 

But we can wait a bit until this is more stable before doing too much of this. 

Great work!